causalml/propensity.py

```diff
@@ -224,14 +223,18 @@ def compute_propensity_score(
 
     p_model.fit(X, treatment)
 
-    if X_pred is None:
-        p = p_model.predict(X)
-    else:
-        p = p_model.predict(X_pred)
+    X0 = X if X_pred is None else X_pred
+
+    try:
+        p = p_model.predict_proba(X0)[:, 1]
+    except AttributeError:
+        logger.info("predict_proba not available, using predict instead")
+        p = p_model.predict(X0)
```

Comment:
Why don't we use `X` for `X_pred` if `X_pred` is none instead of creating a new variable `X0`?

Reply:
Sorry, @ras44. I meant using `X_pred` instead of `X`. 

The rationals are:
1. The code is for prediction, so it makes logical sense to use `X_pred` instead of `X`.
2. Modifying the input within a function other than the optimization purpose is not a good practice.

It's a nitpick and doesn't have any practical implication because, in this case, it won't change the original `X` and create a new variable only within the function, but, in general, there's a case you may unintentionally modify the input (e.g., updating values of numpy arrays or pandas dataframes with indexing).

Reply:
absolutely- I should have caught that 😄